lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt

```diff
@@ -106,20 +103,25 @@ data class InvokeTransaction(
     @SerialName("nonce")
     override val nonce: Felt,
 
+    @SerialName("type")
     override val type: TransactionType = TransactionType.INVOKE,
+
+    // Tx v0
+    @SerialName("contract_address")
+    val contractAddress: Felt = Felt.ZERO,
+
+    // Tx v0
+    @SerialName("entry_point_selector")
+    val entryPointSelector: Felt = Felt.ZERO,
```

Comment:
Yeah, that was my concern too. I finally decided to do it that way, because otherwise we would have multiple dataclasses for multiple transaction types and versions (more so when new versions come out), and I didn't like that idea either.

I guess we could create a data class from which subsequent versions could inherit, but again I'm not sure if this apporach is the best. What do you think?

Reply:
Hmm, v0 is gonna get dropped with regenesis hopefully, so we'll be able to remove this code. However concern about future versions is valid too. Prhaps let's add some base class for the sake of having it for now?